tests/integration/targets/ec2_spot_instance/tasks/main.yaml

```diff
@@ -145,6 +157,35 @@
       launch_spec: '{{ complex_create_result.spot_request.launch_specification }}'
       spot_request_tags: '{{ complex_create_result.spot_request.tags }}'
 
+  - name: Get info about the complex spot instance request created
+    ec2_spot_instance_info:
+      spot_instance_request_ids:
+        - "{{ complex_create_result.spot_request.spot_instance_request_id }}"
+    register: complex_spot_instance_info_result
+
+  - name: Assert that the complex spot request created is open/active and correct keys are set
+    assert:
+      that:
+        - complex_spot_instance_info_result.spot_request[0].state in ['open', 'active']
+        - complex_create_result.spot_request.spot_price == "0.002000"
+        - complex_create_result.spot_request.launch_specification.block_device_mappings[0].ebs.volume_size == 5
+        - complex_create_result.spot_request.launch_specification.block_device_mappings[0].ebs.volume_type == "gp3"
+        - complex_create_result.spot_request.launch_specification.network_interfaces[0].delete_on_termination == True
+
+  - name: Get info about the created spot instance requests and filter result based on provided filters
+    ec2_spot_instance_info:
+      spot_instance_request_ids:
+        - '{{ create_result.spot_request.spot_instance_request_id }}'
+        - '{{ complex_create_result.spot_request.spot_instance_request_id }}'
+      filters:
+        launch.block-device-mapping.device-name: /dev/sdb
```

Comment:
If there's more than one copy of this test running at the same time in CI this can return multiple results (for example, 2 PRs for the same module that get checked at the same time). We should also filter on the unique tag we talked about adding.